distributed/worker_state_machine.py

```diff
@@ -2193,6 +2220,18 @@ def _transition_executing_long_running(
             self._ensure_computing(),
         )
 
+    def _transition_cancelled_long_running(
+        self, ts: TaskState, compute_duration: float, *, stimulus_id: str
+    ) -> RecsInstrs:
+        """This transition also serves resumed(fetch) -> long-running"""
```

Comment:
Not terribly happy about this, but the two alternatives (call it `_transition_generic_long_running`, when it's definitely not generic, and copy-pasting the whole thing to a `_transition_resumed_long_running`) seemed worse.